packages/perseus-core/src/parse-perseus-json/perseus-parsers/radio-widget.test.ts

```diff
@@ -108,6 +108,193 @@ describe("parseRadioWidget", () => {
 
         expect(parse(widget, parseRadioWidget)).toEqual(anySuccess);
     });
+
+    describe("choice ID validation", () => {
+        it("generates default ID when choice ID is an empty string", () => {
+            const widget = {
+                type: "radio",
+                version: {major: 3, minor: 0},
+                graded: true,
+                options: {
+                    choices: [
+                        {
+                            content: "Choice with empty ID",
+                            id: "", // Empty string should trigger fallback
+                        },
+                    ],
+                },
+            };
+
+            const result = parse(widget, parseRadioWidget);
+
+            expect(result).toEqual(
+                success({
+                    ...widget,
+                    options: {
+                        ...widget.options,
+                        choices: [
+                            {
+                                content: "Choice with empty ID",
+                                id: "radio-choice-0", // Should be generated
+                            },
+                        ],
+                    },
+                }),
+            );
+        });
+
+        it("generates default ID when choice ID is whitespace-only", () => {
```

Comment:
Do we actually need to treat whitespace-only IDs as invalid? It doesn't seem like a case we're likely to ever encounter.